### ISSUE TITLE
Regression: ConsulResolver environment variable expansion doesn't work without legacy mode

### ISSUE THREAD

**Issue by woz5999** [datawire/ambassador#3317]
**Describe the bug**
Per https://www.getambassador.io/docs/edge-stack/latest/topics/running/resolvers/ environment variable expansion is supported in the resolver address. However, as of 1.12.1, this doesn't work unless `AMBASSADOR_LEGACY_MODE` is also set. 

```
time="2021-03-29 23:20:11" level=info msg="[ERR] consul.watch: Watch (type: service) errored: Get \"http://$%7BHOST_IP%7D/v1/health/service/test-api?passing=1\": dial tcp: lookup ${HOST_IP}: no such host, retry in 5s" CMD=entrypoint PID=1 oops-i-did-not-pass-context-correctly=true
```
**To Reproduce**
1. Create a ConsulResolver with an environment variable in the `address` field

**Expected behavior**
It should expand the `HOST_IP` variable when querying consul

**Versions (please complete the following information):**
 - Ambassador: 1.12.1

**Comment by woz5999** [comment]
@khussey @kflynn looks like i created a duplicate of #3182

**Comment by khussey** [comment]
The fix for this will be available in the upcoming 1.13 release.
